operation/fhir-operation-validate/src/main/java/com/ibm/fhir/operation/validate/type/ModeType.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.operation.validate.type;
+
+/**
+ * The Mode Type indicates a valid resource validation mode code.
+ * @see <a href="http://hl7.org/fhir/resource-validation-mode">resource validation mode</a>
+ */
+public enum ModeType {
+
+    CREATE("create"),
+    UPDATE("update"),
+    DELETE("delete"),
+    PROFILE("PROFILE");
+
+    private String value;
+
+    /**
+     * Instantiates a new ModeType.
+     *
+     * @param value the value
+     */
+    ModeType(String value) {
+        this.value = value;
+    }
+
+    /**
+     * method to validate a valid resource validation mode code.
+     *
+     * @param value 
+     *      A string that matches one of the allowed ModeType values
+     * @return 
+     *      The corresponding FHIRVersionParam or null if a null value was passed or if no matching resource validation mode code is found 
```

Comment:
In our other Enum classes that have a 'from' method, we throw an IllegalArgumentException if the input String does not match any of the enum values.

I'm not sure if that is the best choice or not, but I mention it because I do like consistency across the codebase.

Reply:
This has been fixed as suggested.  IllegalArgumentException is thrown if the input String is not null and does not match any of the allowed string values.